stock_picking_propagate_scheduled_date/models/stock_move.py

```diff
@@ -2,32 +2,65 @@
 # @author Vincent Van Rossem <vincent.vanrossem@camptocamp.com>
 # License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
 
+from datetime import timedelta
+
 from odoo import fields, models
 
 
 class StockMove(models.Model):
     _inherit = "stock.move"
 
-    def _propagate_date(self, new_date):
-        """Propagate the date of a move to its destination."""
-        already_propagated_ids = self.env.context.get(
-            "date_propagation_ids", set()
-        ) | set(self.ids)
-        self = self.with_context(date_propagation_ids=already_propagated_ids)
+    def write(self, vals):
+        previous_dates = {
+            move.id: move.date for move in self.with_context(prefetch_fields=False)
+        }
```

Comment:
we should compute this only if the condition on line 18 below is true (it is not needed if we are just updating the state, for instance)

Reply:
We need previous dates to get the delta, because propagate is called after super call.

Reply:
I got the idea, sorry for the noise.